src/components/Filters/DateFilter.tsx

```diff
@@ -0,0 +1,101 @@
+import { Key, useState } from "react";
+import { BaseFilter } from "src/components/Filters/BaseFilter";
+import { Filter } from "src/components/Filters/types";
+import { Label } from "src/components/Label";
+import { Css } from "src/Css";
+import { DateField, SelectField, Value } from "src/inputs";
+import { TestIds } from "src/utils";
+import { defaultTestId } from "src/utils/defaultTestId";
+
+export type DateFilterProps<O, V extends Value, DV extends DateFilterValue<V>> = {
+  label: string;
+  operations: O[];
+  getOperationValue: (o: O) => V;
+  getOperationLabel: (o: O) => string;
+  defaultValue?: DV;
+};
+
+export type DateFilterValue<V extends Value> = { op: V; date: Date };
+
+export function dateFilter<O, V extends Key>(
+  props: DateFilterProps<O, V, DateFilterValue<V>>,
+): (key: string) => Filter<DateFilterValue<V>> {
+  return (key) => new DateFilter(key, props);
+}
+
+// Custom option that allows for not selecting an operation
+const anyOption = {} as any;
+
+class DateFilter<O, V extends Key, DV extends DateFilterValue<V>>
+  extends BaseFilter<DV, DateFilterProps<O, V, DV>>
+  implements Filter<DV>
+{
+  render(value: DV, setValue: (value: DV | undefined) => void, tid: TestIds, inModal: boolean, vertical: boolean) {
+    const { label, operations, getOperationValue, getOperationLabel } = this.props;
+    const [focusedEl, setFocusedEl] = useState<"op" | "date" | undefined>();
+    const commonStyles = Css.df.aic.fs1.maxwPx(550).bt.bb.bGray300.$;
+
+    // TODO: Maybe make a `CompoundField` component, which could handle the `display: flex` and any sizing requirements per field and focus states.
```

Comment:
FYI - Schedules V2 has a compound single-select/text input. A beam ticket is likely to be inbound soon.

![image](https://user-images.githubusercontent.com/24765506/137351578-403003c1-253b-465e-9b19-6e512da7a86e.png)

It also has that `x` to clear the field on the right, which doesn't currently exist 🤔

Reply:
Hey @chr1sjf0x , thanks for the heads up! I actually chatted with Dara about this as well, and she showed me this mock. This led me in the direction of refactoring the SelectFieldInput to utilize the TextFieldBase for consistent styling between the two fields. Perhaps as part of the next ticket we can make a more generic "CompoundField" to accomplish what is happening in DateFilter here. 
And yes, we definitely do not have the "X"/clear button yet. Will be on the look out for tickets! Thanks for the review!